jfix-armeria-aggregating-profiler/src/test/kotlin/ru/fix/armeria/aggregating/profiler/ProfiledHttpClientTest.kt

```diff
@@ -346,6 +346,34 @@ internal class ProfiledHttpClientTest {
         }
     }
 
+    @Test
+    suspend fun `non-200 http status is valid`() {
+        val profiler = AggregatingProfiler()
+        val profilerReporter = profiler.createReporter()
+        val mockServer = ArmeriaMockServer {
+            service(ErrorProfilingTest.TEST_PATH) { _, _ ->
+                HttpResponse.of(HttpStatus.FOUND)
+            }
+        }
+        mockServer.start()
+        val client = WebClient
+            .builder(mockServer.httpUri())
+            .decorator(ProfiledHttpClient.newDecorator(profiler) { it == HttpStatus.FOUND })
+            .build()
+        try {
+            client.get(ErrorProfilingTest.TEST_PATH).aggregate().await()
+        } catch (e: Exception) {
+            logger.error(e)
+        }
+        eventually(1.seconds) {
+            assertSoftly(profilerReporter.buildReportAndReset()) {
+                profiledCallReportWithNameEnding(Metrics.HTTP_CONNECT) shouldNotBe null
```

Comment:
as I have discovered during tests stabilization, it is better to assert each metric independently by building separate profiler reports. Otherwise next buildReportAndReset could reset value of metric that reached required value due to other metrics still not reached it

Reply:
also same comment about `shouldNotBe null` is too poor assertion

Reply:
done